packages/@adobe/spectrum-css-temp/components/table/skin.css

```diff
@@ -160,7 +160,24 @@ tbody.spectrum-Table-body {
   }
 
   &.spectrum-Table-row--highlightSelection {
-    border-bottom-color: var(--spectrum-global-color-blue-500);
+    &.spectrum-Table-row--firstRow {
+      border-block-start: 1px solid transparent;
```

Comment:
a full box shadow replacement of the row borders resulted in non curved corners that matched the container so opted to still use border-top/bottom still.